trace-utils/src/tracer_payload.rs

```diff
@@ -106,14 +105,140 @@ impl TracerPayloadCollection {
     }
 }
 
+pub type TracerPayloadChunkProcessor<'a> = Box<dyn Fn(&mut pb::TraceChunk, usize) + 'a>;
```

Comment:
The proper way to implement this is having a `trait TracerPayloadChunkProcessor { fn process(&mut pb::TraceChunk, usize); }` and adding a `PhantomData<T>` to TracerPayloadParams with `T: TracerPayloadChunkProcessor`.

Reply:
I meant _allows_, not avoids.

Saving an extra memory allocation and call mostly. Nothing major, just idiomatic and a tiny bit better.

Reply:
Probably better to prefer static dispatch. Using boxed fn is a bit like Virtual (dynamic) method dispatch in C++ - useful when polymorphism is required but adding overhead. 

Since this is part of trace chunk processing that can happen in a tight loop. 

In addition to extra instructions it will have higher chance to incur a page fault whenever processor is called. 
Ideally we should keep all code needed to do chunk processing within the CPU cache when possible.

Reply:
@bwoebi Ah ok, that makes more sense. 

I agree, using dynamic dispatch is not optimal. I was just trying to limit the blast radius of the refactor and maintain the same function signature for `trace_utils::collect_trace_chunks`.

I didn't want to commit the changes to this branch yet as I still need to do a bunch of cleanup as well as fix tests. [But is this what you had in mind](https://github.com/DataDog/libdatadog/pull/534/files)? It uses a trait as you suggested and avoids the need for any dynamic dispatch. There is no need for `PhantomData` though. If this makes sense, I'll incorporate it into this PR.

Reply:
Yes, that works too :-)